src/tsolvers/egraph/Enode.h

```diff
@@ -48,8 +47,32 @@ struct ERef {
 };
 static struct ERef ERef_Undef = {INT32_MAX};
 
+struct UseVectorIndex {
+    int32_t x;
+    inline friend bool operator== (const UseVectorIndex& a1, const UseVectorIndex& a2) {return a1.x == a2.x; }
+    inline friend bool operator!= (const UseVectorIndex& a1, const UseVectorIndex& a2) {return a1.x != a2.x; }
+    static UseVectorIndex NoIndex;
+    static UseVectorIndex DuplicateIndex;
+};
+
+
+
+static_assert(sizeof(ERef) == sizeof(UseVectorIndex));
+
 inline void swap(ERef & y, ERef & z) { ERef tmp = y; y = z; z = tmp; }
 
+class ERefSpan {
```

Comment:
Let's have the generic span, i.e., `template<typename T> Span {...}` with `T const * begin() const` and `T const * end() const`?

Reply:
I have generalized this span.

Reply:
Yes, I like it 🥇